File: Examples/VisualBasic/Exporting-DGN/ExportDGNAsPartofDWG.vb

Namespace Exporting_DGN

    Public Class ExportDGNAsPartofDWG

        Public Shared Sub Run()
            
            Try
                'ExStart: ExportDGNAsPartofDWG

                ' Input and Output file paths
                Dim fileName As String = "BlockRefDgn.dwg"
                Dim outPath As String = fileName & Convert.ToString(".pdf")

                ' Create an instance of PdfOptions class as we are exporting the DWG file to PDF format
                Dim exportOptions As New Aspose.CAD.ImageOptions.PdfOptions()

                ' Load any existing DWG file as image and convert it to CadImage type
                Using cadImage As Aspose.CAD.FileFormats.Cad.CadImage = DirectCast(Image.Load(fileName), Aspose.CAD.FileFormats.Cad.CadImage)
                    ' Go through each entity inside the DWG file
                    For Each baseEntity As Aspose.CAD.FileFormats.Cad.CadObjects.CadBaseEntity In cadImage.Entities
                        ' Check if entity is an image definition
                        If baseEntity.TypeName = Aspose.CAD.FileFormats.Cad.CadConsts.CadEntityTypeName.DGNUNDERLAY Then
                            Dim dgnFile As Aspose.CAD.FileFormats.Cad.CadObjects.CadDgnUnderlay = DirectCast(baseEntity, Aspose.CAD.FileFormats.Cad.CadObjects.CadDgnUnderlay)

                            ' Get external reference to object
                            System.Console.WriteLine(dgnFile.UnderlayPath)
                        End If
                    Next

                    ' Define settings for CadRasterizationOptions object
		exportOptions.VectorRasterizationOptions = New Aspose.CAD.ImageOptions.CadRasterizationOptions() With { _
			Key .PageWidth = 1600, _
			Key .PageHeight = 1600, _
			Key .CenterDrawing = True, _
			Key .Layouts = New String() {"Model"}, _
			Key .ScaleMethod = Aspose.CAD.FileFormats.Cad.ScaleType.None, _
			Key .BackgroundColor = Color.Black, _
			Key .DrawType = Aspose.CAD.FileFormats.Cad.CadDrawTypeMode.UseObjectColor _
		}

                    ' Export the DWG to PDF by calling Save method
                    cadImage.Save(outPath, exportOptions)
                    'ExEnd:ExportDGNAsPartofDWG

                End Using
            Catch ex As Exception
                Console.WriteLine("Please use valid input file." + ex.Message)
            End Try


        End Sub

    End Class

End Namespace

File: Examples/CSharp/Exporting-DGN/ExportDGNAsPartofDWG.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Aspose.CAD;


namespace Aspose.CAD.Examples.CSharp.Exporting_DGN
{
    class ExportDGNAsPartofDWG
    {

        public static void Run()
        {
            try
            {
                //ExStart:ExportDGNAsPartofDWG

                // Input and Output file paths
                string fileName = "BlockRefDgn.dwg";
                string outPath = fileName + ".pdf";

                // Create an instance of PdfOptions class as we are exporting the DWG file to PDF format
                Aspose.CAD.ImageOptions.PdfOptions exportOptions = new Aspose.CAD.ImageOptions.PdfOptions();

                // Load any existing DWG file as image and convert it to CadImage type
                using (Aspose.CAD.FileFormats.Cad.CadImage cadImage = (Aspose.CAD.FileFormats.Cad.CadImage)Image.Load(fileName))
                {
                    // Go through each entity inside the DWG file
                    foreach (Aspose.CAD.FileFormats.Cad.CadObjects.CadBaseEntity baseEntity in cadImage.Entities)
                    {
                        // Check if entity is an image definition
                        if (baseEntity.TypeName == Aspose.CAD.FileFormats.Cad.CadConsts.CadEntityTypeName.DGNUNDERLAY)
                        {
                            Aspose.CAD.FileFormats.Cad.CadObjects.CadDgnUnderlay dgnFile = (Aspose.CAD.FileFormats.Cad.CadObjects.CadDgnUnderlay)baseEntity;

                            // Get external reference to object
                            System.Console.WriteLine(dgnFile.UnderlayPath);
                        }
                    }

                    // Define settings for CadRasterizationOptions object
                    exportOptions.VectorRasterizationOptions = new Aspose.CAD.ImageOptions.CadRasterizationOptions()
                    {
                        PageWidth = 1600,
                        PageHeight = 1600,
                        CenterDrawing = true,
                        Layouts = new string[] { "Model" },
                        AutomaticLayoutsScaling = false,
                        NoScaling = true,
                        BackgroundColor = Color.Black,
                        DrawType = Aspose.CAD.FileFormats.Cad.CadDrawTypeMode.UseObjectColor
                    };

                    //Export the DWG to PDF by calling Save method
                    cadImage.Save(outPath, exportOptions);
                }
                //ExEnd:ExportDGNAsPartofDWG

            }
            catch (Exception ex)
            {
                Console.WriteLine("Please use valid input file." + ex.Message);
            }

        }

    }
}
